src/compiler/checker.ts

```diff
@@ -3645,6 +3645,19 @@ namespace ts {
             return result || emptyArray;
         }
 
+        function getNamedOrIndexSignatureMembers(members: SymbolTable): Symbol[] {
+            const result = getNamedMembers(members);
+            const index = getIndexSymbolFromSymbolTable(members);
+            if (!index) {
+                return result;
+            }
+            if (!result.length) {
+                return [index];
+            }

```

Comment:
If this is done to avoid mutating `emptyArray` then maybe it's clearer to test `result === emptyArray`?  (Or if not, then a comment?)